src/main/java/net/mcreator/element/converter/fv13/ProcedureStringsFixer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2020 Pylo and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.converter.fv13;
+
+import com.google.gson.JsonElement;
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.converter.IConverter;
+import net.mcreator.element.types.Procedure;
+import net.mcreator.workspace.Workspace;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ProcedureStringsFixer implements IConverter {
+
+	private static final Logger LOG = LogManager.getLogger("ProcedureStringsFixer");
+
+	@Override
+	public GeneratableElement convert(Workspace workspace, GeneratableElement input, JsonElement jsonElementInput) {
+		Procedure procedure = (Procedure) input;
+		try {
+			procedure.procedurexml = fixXML(procedure.procedurexml);
+		} catch (Exception e) {
+			LOG.warn("Failed to fix string dependency for procedure " + input.getModElement().getName());
+		}
+		return procedure;
+	}
+
+	@Override public int getVersionConvertingTo() {
+		return 13;
+	}
+
+	protected String fixXML(String xml) throws Exception {
+		DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+		Document doc = dBuilder.parse(new InputSource(new StringReader(xml)));
+		doc.getDocumentElement().normalize();
+
+		NodeList nodeList = doc.getElementsByTagName("block");
```

Comment:
The code lines below this will not fix and add a default value.

Reply:
The commit is called "Started ..." for a reason, it's not done yet I'm waiting for Goldorion's help.

Reply:
Then convert PR to draft to indicate it is not ready yet.

Reply:
Will be fixed as soon as Goldorion helps me...

Reply:
Check how it was done for adding Source/event entity block, minimal changes would be needed I believe.

Reply:
I will mark this resolved as this part of the code was not actually code review ready, I just didn't know it.